### DISCUSSION TITLE
[Question] Why does parse_arguments function convert overrides to YAML string before return?

### DISCUSSION THREAD

**Discussion by kokamido** [speechbrain/speechbrain#1636] [Q&A] [answered]
Hi. The docstring of the parse_arguments function describes `overrides` as a dictionary:
https://github.com/speechbrain/speechbrain/blob/e14d2c15874a22df018edf441af62b105b33ed39/speechbrain/core.py#L143
But parse_arguments converts `overrides` to a YAML string before return.
https://github.com/speechbrain/speechbrain/blob/e14d2c15874a22df018edf441af62b105b33ed39/speechbrain/core.py#L279
What is the reason for this conversion?

**Comment by Gastron** [comment]
Good point! This must be due to coupling between that and `load_hyperpyyaml`, https://speechbrain.readthedocs.io/en/latest/API/hyperpyyaml.core.html#hyperpyyaml.core.load_hyperpyyaml. I bet this could be decoupled - maybe the conversion to YAML string could happen in load_hyperpyyaml, which claims to accepts both mapping and string as input. 

@pplantinga have you got time to chime in?

**Comment by pplantinga** [comment, marked as answer]
The documentation is incorrect here, the type of `overrides` is clearly a string. The `_convert_to_yaml()` function takes a string formatted as command line arguments (i.e. `--arg=val` or `--arg val`) and converts it to yaml format. This provides a little more flexibility than converting it to dict, because you can leverage any hyperpyyaml tags (e.g. `!new:` or `!name:`).
